### ISSUE TITLE
Update rincon_connector deps: hyper -> 0.12, hyper-tls -> 0.3, native-tls -> 0.2 (fixes openssl 0.9 dependency)

### ISSUE THREAD

**Issue by jeffvandyke** [innoave/rincon#11]
Openssl 0.9 is breaking on a growing number of Linux systems [see this issue](https://github.com/sfackler/rust-openssl/issues/994). hyper-tls 0.3 is the first to depend on native-tls 0.2, which is the first native-tls version to drop openssl 0.9. Hyper-tls, as far as I can tell, needs hyper 0.12 to work.

We currently use hyper 0.11, and hyper 0.12 is quite an upgrade with a few breaking changes ([release notes](https://github.com/hyperium/hyper/releases/tag/v0.12.0)).

An upgrade also breaks the hyper-timeout dependency, which as far as I can tell, needs hyper 0.11 to work (I've filed another issue [here](https://github.com/hjr3/hyper-timeout/issues/5)). An idea would be to drop hyper-timeout in favor of using `tokio_core::reactor` to timeout from outside hyper inside of `impl Execute for JsonHttpConnection -> execute()` inside `rincon_connector/src/http/mod.rs`, following advice on hyper's issue ["Request timeouts"](https://github.com/hyperium/hyper/issues/1097). It also looks like Hyper is working on client timeouts at [this issue](https://github.com/hyperium/hyper/issues/1628) (last update April 2019).

Another annoying change is that the typed headers (like `ContentType::json()`, most stuff from `hyper::header`) were removed, though they might be available in the [headers crate](https://crates.io/crates/headers), which is related to hyper and has good usage.

The biggest decision is what to do about timeouts, and I don't care for the idea of waiting for other projects to fix issues.

So basically, `cargo test` doesn't work on a fresh Rincon clone on my Manjaro Linux machine right now, and this is an explanation of the rabbit hole I went down to find out why. Bit of a mess, but there you go. High hopes!

**Comment by jeffvandyke** [comment]
As something to keep an eye on, Hyper is working on 0.13, but it's uncertain when they'll be finished:[Hyper 0.13 milestone](https://github.com/hyperium/hyper/milestone/5)

**Comment by jeffvandyke** [comment]
So it looks like @hjr3 is getting ready for a new release on hyper-timeout to hyper 0.12 - [pull request here](https://github.com/hjr3/hyper-timeout/pull/6), which would resolve the timeout question here and allow upgrading Rincon and keep the same dependency. I might try this out as a learning experiment. Would you be open to a pull request to Rincon to upgrade to hyper 0.12?
